tests/unit/sagemaker/workflow/test_condition_step.py

```diff
@@ -60,3 +77,26 @@ def test_condition_step():
         },
     }
     assert cond_step.properties.Outcome.expr == {"Get": "Steps.MyConditionStep.Outcome"}
+
+
+def test_pipeline(sagemaker_session):
+    param = ParameterInteger(name="MyInt", default_value=2)
+    cond = ConditionEquals(left=param, right=1)
+    custom_step1 = CustomStep("IfStep")
+    custom_step2 = CustomStep("ElseStep")
+    step_cond = ConditionStep(
+        name="CondStep",
+        conditions=[cond],
+        if_steps=[custom_step1],
+        else_steps=[custom_step2],
+    )
+    pipeline = Pipeline(
+        name="MyPipeline",
+        steps=[step_cond],
+        sagemaker_session=sagemaker_session,
+        parameters=[param],
+    )
+    adjacency_list = PipelineGraph.from_pipeline(pipeline).adjacency_list
+    assert ordered(adjacency_list) == ordered(
+        {"CondStep": ["IfStep", "ElseStep"], "IfStep": [], "ElseStep": []}
+    )
```

Comment:
From visualization perspective, seems it's not able to tell which steps, in adjacency list of "CondStep", are in IF branch and which are in ELSE branch.
Please correct me if I'm wrong

Reply:
Nic's design doc uses the following adjacency list format:
```
[
        {"StepName": 'AbaloneProcess',
         "OutBoundEdges": [
            {
                "nextStepName": "AbaloneTrain",
                "edgeLabel": None                <<<<<<<<<<< indicating the IF or ELSE branch
            },
            {
                "nextStepName": "AbaloneEval",
                "edgeLabel": None
            }
         ]
        },
        {"StepName": 'AbaloneMSECond',
         "OutBoundEdges": [
            {
                "nextStepId": "AbaloneRegisterModel",
                "edgeLabel": "true"
            },
            {
                "nextStepId": "AbaloneCreateModel",
                "edgeLabel": "true"
            },
            {
                "nextStepId": "AbaloneMSEFail",
                "edgeLabel": "false"
            }
         ]
        },
```

Can we also apply it here?

Reply:
This format can be derived using a combination of the current adjacency list format and the `pipeline.steps` field.
Same thing applies to the condition step if-else branches. The goal of the adjacency list to give parent-child step relationship info. Nic can use the adjacency list + pipeline.steps to decipher whether the child of a condition step is a if-child or else-child.